modules/social_features/social_group/src/Tests/Unit/CurrentGroupServiceTest.php

```diff
@@ -0,0 +1,161 @@
+
+  /**
+   * The mocked ContextProviderInterface.
+   *
+   * @var \Drupal\Core\Plugin\Context\ContextProviderInterface|\PHPUnit\Framework\MockObject\MockObject
+   */
+  private ContextProviderInterface|MockObject $groupRouteContext;
+
+  /**
+   * The service under test.
+   *
+   * @var \Drupal\social_group\CurrentGroupService
+   */
+  private CurrentGroupService $currentGroupService;
+
+  /**
+   * Set up the test case.
+   */
+  protected function setUp(): void {
+    parent::setUp();
+
+    $this->entityTypeManager = $this->createMock(EntityTypeManagerInterface::class);
+    $this->groupRouteContext = $this->createMock(ContextProviderInterface::class);
+
+    $this->currentGroupService = new CurrentGroupService(
+      $this->entityTypeManager,
+      $this->groupRouteContext
+    );
+  }
+
+  /**
+   * Helper function to create a mock context.
+   *
+   * @param mixed $value
+   *   The value returned by typeDataInterface.
+   *
+   * @return \Drupal\Core\Plugin\Context\ContextInterface
+   *   The updated mock context.
+   */
+  private function createContextWithValue(mixed $value): ContextInterface {
+    $typeDataInterface = $this->createMock(TypedDataInterface::class);
+    $typeDataInterface->expects($this->once())
+      ->method('getValue')
+      ->willReturn($value);
+
+    $context = $this->createMock(ContextInterface::class);
+    $context->expects($this->once())
+      ->method('getContextData')
+      ->willReturn($typeDataInterface);
+
+    return $context;
+  }
+
+  /**
+   * Test that fromRunTimeContexts() returns NULL when no group context is set.
+   */
+  public function testFromRunTimeContextsNoGroupContext(): void {
+    $this->groupRouteContext
+      ->expects($this->once())
+      ->method('getRuntimeContexts')
+      ->willReturn([]);
+
+    $group = $this->currentGroupService->fromRunTimeContexts();
+
+    $this->assertNull($group, 'No group context should return NULL.');
+  }
+
+  /**
+   * Test that fromRunTimeContexts() returns a GroupInterface when available.
+   */
+  public function testFromRunTimeContextsWithGroup(): void {
+    $group = $this->createMock(GroupInterface::class);
+    $context = $this->createContextWithValue($group);
+
+    $this->groupRouteContext
+      ->expects($this->once())
+      ->method('getRuntimeContexts')
+      ->willReturn(['group' => $context]);
+
+    $result = $this->currentGroupService->fromRunTimeContexts();
+
+    $this->assertSame($group, $result, 'Group should be returned when found in context.');
+  }
+
+  /**
+   * Test that fromRunTimeContexts() loads a group by ID.
+   */
+  public function testFromRunTimeContextsWithGroupId(): void {
+    $groupId = 1;
+    $group = $this->createMock(GroupInterface::class);
+
+    $context = $this->createContextWithValue($groupId);
+
+    $this->groupRouteContext
+      ->expects($this->once())
+      ->method('getRuntimeContexts')
+      ->willReturn(['group' => $context]);
```

Comment:
This mock violates the contract of the group services' getRuntimeContexts` as explained in a previous comment.